File: app/src/main/java/com/system/m4/views/home/MainContract.java
package com.system.m4.views.home;

import com.system.m4.views.vos.TagVO;
import com.system.m4.views.vos.Transaction;

/**
 * Created by eferraz on 29/07/17.
 * For M4
 */

public interface MainContract {

    interface View{

        void requestTransactionManagerDialog();

        void showTransactionDialog(TagVO vo);

        void showTransactionDialog(Transaction vo);

        void setMainTitle(String title);
    }

    interface Presenter {

        void requestTransactionManager();

        void requestTransactionDialog(TagVO vo);

        void configureTitle(int position);
    }
}

File: app/src/main/java/com/system/m4/kotlin/contracts/MainContract.kt
package com.system.m4.kotlin.contracts

import com.system.m4.labs.vos.TagVO
import com.system.m4.labs.vos.TransactionVO

/**
 * Created by eferraz on 29/07/17.
 * For M4
 */

interface MainContract {

    interface View {

        fun requestTransactionManagerDialog()

        fun showTransactionDialog(vo: TagVO)

        fun showTransactionDialog(vo: TransactionVO)

        fun setMainTitle(title: String)
    }

    interface Presenter {

        fun requestTransactionManager()

        fun requestTransactionDialog(vo: TagVO)

        fun configureTitle(position: Int)
    }
}
